Sources/MapboxMaps/Foundation/Extensions/Core/ResourceOptions.swift

```diff
@@ -41,90 +84,122 @@ extension ResourceOptions {
+                                                    withIntermediateDirectories: true,
+                                                    attributes: nil)
 
-        try FileManager.default.createDirectory(at: cacheDirectoryURL,
-                                                withIntermediateDirectories: true,
-                                                attributes: nil)
+            cacheDirectoryURL.setTemporaryResourceValue(true, forKey: .isExcludedFromBackupKey)
 
-        cacheDirectoryURL.setTemporaryResourceValue(true, forKey: .isExcludedFromBackupKey)
-
-        return cacheDirectoryURL.appendingPathComponent("ambient_cache.db")
+            return cacheDirectoryURL.appendingPathComponent("ambient_cache.db")
+        } catch {
+            fatalError("Failed to create cache directory: \(error)")
+        }
     }
+}
 
+extension ResourceOptions: Hashable {
     /// :nodoc:
-    public override func isEqual(_ object: Any?) -> Bool {
-
-        guard let other = object as? ResourceOptions else {
-            return false
-        }
-
-        guard type(of: self) == type(of: other) else {
-            return false
-        }
-
-        return (accessToken == other.accessToken)
-            && ((baseURL == other.baseURL)
-                    || (baseURL == nil)
-                    || (other.baseURL == nil))
-            && (cachePath == other.cachePath)
-            && (assetPath == other.assetPath)
-            && (cacheSize == other.cacheSize)
-            && ((tileStore == other.tileStore)
-                    || (tileStore == nil)
-                    || (other.tileStore == nil))
-            && (tileStoreUsageMode == other.tileStoreUsageMode)
+    public static func == (lhs: Self, rhs: Self) -> Bool {
+        return (lhs.accessToken == rhs.accessToken)
+            && ((lhs.baseURL == rhs.baseURL)
+                    || (lhs.baseURL == nil)
+                    || (rhs.baseURL == nil))
+            && (lhs.cachePathURL == rhs.cachePathURL)
+            && (lhs.assetPathURL == rhs.assetPathURL)
+            && (lhs.cacheSize == rhs.cacheSize)
+            && ((lhs.tileStore == rhs.tileStore)
+                    || (lhs.tileStore == nil)
+                    || (rhs.tileStore == nil))
+            && (lhs.tileStoreUsageMode == rhs.tileStoreUsageMode)
     }
 
     /// :nodoc:
-    open override var hash: Int {
-        var hasher = Hasher()
+    public func hash(into hasher: inout Hasher) {
         hasher.combine(accessToken)
         hasher.combine(baseURL)
-        hasher.combine(cachePath)
-        hasher.combine(assetPath)
+        hasher.combine(cachePathURL)
+        hasher.combine(assetPathURL)
         hasher.combine(cacheSize)
         hasher.combine(tileStore)
         hasher.combine(tileStoreUsageMode)
-        return hasher.finalize()
+    }
+}
+
+extension ResourceOptions: CustomStringConvertible, CustomDebugStringConvertible {
+    private func redactedAccessToken() -> String {
```

Comment:
If we do add a custom AccessToken type, then this description logic would move there too.